### PR TITLE
MBS-10659: Add prompt admin settings

### PR DESCRIPTION
**Pull request by HerrZerman**
userprompt setting removed — single template covers everything via Mustache ({{#mode_topic}}, {{#mode_story}}, {{#sendexistingquestions}} etc.)
New render_prompt_template() method replaces hardcoded prompt building
db/upgrade.php: non-breaking upgrade step writes new default if setting is empty
Unit tests extended for all modes and edge cases

### CODE REVIEW

File: classes/local/question_generator.php [hunk 1]
```diff
@@ -51,6 +51,121 @@ public function __construct(
...
+        // block is included and never touches the actual content.
+        $mustache = new \Mustache_Engine(['escape' => fn($value) => $value]);
+        $rendered = $mustache->render($template, [
+            'sendexistingquestions' => $sendexistingquestions,
+            'primer' => '{{primer}}',
+            'instructions' => '{{instructions}}',
+            'example' => '{{example}}',
+            'existingquestionsjson' => '{{existingquestionsjson}}',
+        ]);
+        return self::substitute_placeholders($rendered, $values);
```
Comment (PhMemmel) [inline classes/local/question_generator.php:145]: Well this is supercomplicated and not necessary. I know that in an earlier feedback I said we shouldn't use mustache here, because escaping could be an issue. However, you now use mustache anyway, replace it with the placeholders again just to substitute them afterwards in a separate function which just doesn't make sense.

Two options:
- Let go off the mustache completely. Make a manual preg_replace that strips out everything between `{{#sendexistingquestions}} and {{^sendexistingquestions}}` if `$sendexistingquestions` is false and use the manual `substitute_placeholders` function. Advantage: mustache will be dropped in moodle at some point (even if this will take a long time), code is pretty readable and obvious.
- Use mustache to the full extent: Then please use \Mustache\Engine instead of \Mustache_Engine (the latter one is deprecated):
```
        $mustache = new \Mustache_Engine(['escape' => fn($value) => $value]);
        return $mustache->render($template, [
            'sendexistingquestions' => $sendexistingquestions,
            'primer' => $values['primer'],
            'instructions' => $values['instructions'],
            'example' => $values['example'],
            'existingquestionsjson' => $values['existingquestionsjson'],
        ]);
```

Please also change the function phpdocs accordingly

File: tests/local/question_generator_test.php [hunk 2]
```diff
@@ -30,15 +30,124 @@
...
+        $existingjson = '[{"title":"Q1","question_text":"Some text"}]';
+
+        $values = [
+            'primer' => $primer,
+            'instructions' => $instructions,
+            'example' => $example,
+            'existingquestionsjson' => '',
+        ];
+
+        return [
```
Comment (PhMemmel) [inline tests/local/question_generator_test.php:83]: you also want to have some more cases in which "strange" characters are included. Special chars like `#'ßÄ` and whatever you can imagine, also html and latex code, json strings etc.

File: version.php [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 $plugin->component = 'qbank_questiongen';
 $plugin->release = '2.3';
-$plugin->version  = 2026050800;
+$plugin->version  = 2026050801;
```
Comment (PhMemmel) [inline version.php:29]: Please rebase your branch against main. This will conflict at this line, please use this to update it to the version string matching "today".

File: classes/local/question_generator.php [hunk 4]
```diff
@@ -96,35 +211,35 @@ public function generate_question(stdClass $dataobject, bool $sendexistingquesti
...
+                    'example' => $example,
+                    'existingquestionsjson' => $existingquestionsjson,
+                ]
+            );
+
+            // Select the user message template depending on the generation mode. Topic mode lets the LLM use its own
+            // training data, whereas story/course contents mode restricts the LLM to the provided content.
+            if ($dataobject->mode === story_form::QUESTIONGEN_MODE_TOPIC) {
+                $usermessagetemplate = get_config('qbank_questiongen', 'usermessagetopic');
+                if (empty($usermessagetemplate)) {
```
Comment (PhMemmel) [inline classes/local/question_generator.php:233]: I don't think we want this extra guard here. If the config is empty, then someone has emptied it intentionally. So please just use the setting here and also for the other settings. If they are empty, than this has been intentional. Moodle's setting API should be responsible for setting the proper default values.

**Review by PhMemmel — CHANGES_REQUESTED** [review]
